packages/mantine/src/components/table/Table.tsx

```diff
@@ -24,58 +24,74 @@ import {TableHeader} from './TableHeader';
-        width: '100%',
-        '& td:first-of-type': {
-            paddingLeft: theme.spacing.xl,
+interface TableStylesParams {
+    hasHeader: boolean;
+    multiRowSelectionEnabled: boolean;
+}
+
+const useStyles = createStyles<string, TableStylesParams>((theme, {hasHeader, multiRowSelectionEnabled}) => {
+    const rowBackgroundColor =
+        theme.colorScheme === 'dark'
+            ? theme.fn.rgba(theme.colors[theme.primaryColor][7], 0.2)
+            : theme.colors[theme.primaryColor][0];
+    return {
+        table: {
+            width: '100%',
+            '& td:first-of-type, th:first-of-type > *': {
+                paddingLeft: theme.spacing.xl,
+            },
         },
-    },
 
-    header: {
-        position: 'sticky',
-        top: hasHeader ? 69 : 0,
-        backgroundColor: theme.colorScheme === 'dark' ? theme.black : theme.white,
-        transition: 'box-shadow 150ms ease',
-        zIndex: 12, // skeleton is 11
-        '& tr th:first-of-type button': {
-            paddingLeft: theme.spacing.xl,
+        header: {
+            position: 'sticky',
+            top: hasHeader ? 69 : 0,
+            backgroundColor: theme.colorScheme === 'dark' ? theme.black : theme.white,
+            transition: 'box-shadow 150ms ease',
+            zIndex: 12, // skeleton is 11
+
+            '&::after': {
+                content: '""',
+                position: 'absolute',
+                left: 0,
+                right: 0,
+                bottom: 0,
+                borderBottom: `1px solid ${theme.colors.gray[2]}`,
+            },
         },
-        '& tr th:first-of-type div': {
-            paddingLeft: theme.spacing.xl,
+
+        rowSelected: {
+            backgroundColor: multiRowSelectionEnabled ? undefined : rowBackgroundColor,
         },
 
-        '&::after': {
-            content: '""',
-            position: 'absolute',
-            left: 0,
-            right: 0,
-            bottom: 0,
-            borderBottom: `1px solid ${theme.colors.gray[2]}`,
+        rowSelectionCheckboxCell: {
+            verticalAlign: 'middle',
         },
-    },
 
-    rowSelected: {
-        ref: getRef('rowSelected'),
-    },
+        rowCollapsibleButtonCell: {
+            textAlign: 'right',
+        },
 
-    row: {
-        [`&:hover, &.${getRef('rowSelected')}`]: {
-            backgroundColor:
-                theme.colorScheme === 'dark'
-                    ? theme.fn.rgba(theme.colors[theme.primaryColor][7], 0.2)
-                    : theme.colors[theme.primaryColor][0],
+        row: {
+            '&:hover': {
+                backgroundColor: rowBackgroundColor,
+            },
         },
-    },
-}));
+    };
+});
 
 interface TableProps<T> {
     /**
      * Data to display in the table
      */
     data: T[];
+    /**
+     * Defines how each row is uniquely identified. It is highly recommended that you specify this prop to an ID that makes sense.
+     */
+    getRowId?: CoreOptions<T>['getRowId'];
```

Comment:
I was wondering, what happen when you do set this prop with the multiselect? Do we have a default return value (or we assume that if you enable the multi select, it should be mandatory)?

Reply:
> If you need to identify individual rows that are originating from any server-side operations, it's suggested you use this function to return an ID that makes sense regardless of network IO/ambiguity eg. a userId, taskId, database ID field, etc.

https://tanstack.com/table/v8/docs/api/core/table#getrowid